### PR TITLE
9 subissue 11 backend skeleton

### LINKED ISSUE #9: [SubIssue 1.1] Backend Skeleton

### PR DESCRIPTION
**Pull request by MMachadoUofT**
Considering this is my pull request, let me know what you want to see next time. Thankfully this is pretty bare-bones, just creating the entity interfaces so people can write their code without having IntelliJ scream at them.

### CODE REVIEW

File: src/main/java/entityLayer/CommonCurriculum.java [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+    @Override
+    public TaskTree getTask(int taskID) {
+        return null;
+    }
+
+    @Override
+    public TimeBlockManager getThisWeekSchedule() {
+        return null;
+    }
 }
```
Comment (bmguiler) [inline src/main/java/entityLayer/CommonCurriculum.java:53]: also a method getName so we know which curriculum we are dealing with within the schedule

File: src/main/java/entityLayer/CommonSchedule.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    @Override
+    public boolean removeAvailability(TimeBlock unavailableBlock) {
+        return false;
+    }
+
+    @Override
+    public Curriculum getCurriculum(int curriculumID) {
+        return null;
+    }
+}
```
Comment (bmguiler) [inline src/main/java/entityLayer/CommonSchedule.java:59]: method for changeAvailability

File: src/main/java/entityLayer/CommonTimeBlockManager.java [hunk 3]
```diff
@@ -1,4 +1,20 @@
...
+    @Override
+    public void addTimeBlock(TimeBlock timeBlock) {
+
+    }
+
+    @Override
+    public boolean removeTimeBlock(TimeBlock timeBlock) {
+        return false;
+    }
 }
```
Comment (bmguiler) [inline src/main/java/entityLayer/CommonTimeBlockManager.java:50]: we talked about a method for checking to make sure that a time block lies within the users availability, not sure if that will still be useful or not

File: src/main/java/entityLayer/CommonUser.java [hunk 4]
```diff
@@ -1,4 +1,23 @@
...
+    @Override
+    public void setEmail() {
+
+    }
+
+    @Override
+    public void setPassword() {
+
+    }
 }
```
Comment (bmguiler) [inline src/main/java/entityLayer/CommonUser.java:81]: will the user have a username too or just an email?

**Review by bmguiler — DISMISSED** [review]
Hey Marcial this is really great. I know you said that this is bare bones just to get us started which I really appreciate, I just added a few comments of some other methods that we talked about implementing in some of the classes

File: src/main/java/entityLayer/CommonUser.java (hunk 4 above)
Comment (bmguiler) [reply src/main/java/entityLayer/CommonUser.java:81]: oh also a method for getAvailability

**Review by dadler83 — CHANGES_REQUESTED** [review]
For all of the classes there should be file doc strings and method doc strings, and for all of the interfaces there should be file docstrings that reference the classes they interact with.

Also, you should add your name as an author to all of the file doc strings along with a date created, and last updated. This lets us more easily track who has worked on what, the function of every class / interface, and when these changes were made.

**Review by Eggathin — DISMISSED** [review]

File: src/main/java/entityLayer/CommonUser.java (hunk 4 above)
Comment (MMachadoUofT) [reply src/main/java/entityLayer/CommonUser.java:81]: I'm looking into adding all of these now; However, I'm gonna hold off on creating a getAvailability method for the user, instead having client code first get the _schedule_, then calling the Schedule's getAvailability method (which, looking now, I'm having to implement. Oops. Guess that's why we do code reviews!)

File: src/main/java/entityLayer/CommonSchedule.java (hunk 2 above)
Comment (MMachadoUofT) [reply src/main/java/entityLayer/CommonSchedule.java:59]: I'll go ahead and put that in, but I'm going to hold off a while on actually implementing. We've got 10 days until the deadline as of me writing this reply, and creating a "changeavailability" method will require also changing every single curriculum's TimeManager objects to fit into the new availability, which will probably also have the user have to review it, etc, etc.While I'd love to implement that functionality, it just isn't as pressing as our more foundational functionalities, and poses way too many complications for us to begin considering implementing it this early into the process.

I'll type it into the interface just to have it, but I'll make a note in the documentation that it has yet to be implemented, and that the implementation will be something to be decided on later.

File: src/main/java/entityLayer/CommonCurriculum.java (hunk 1 above)
Comment (MMachadoUofT) [reply src/main/java/entityLayer/CommonCurriculum.java:53]: Got it!

File: src/main/java/entityLayer/CommonTimeBlockManager.java (hunk 3 above)
Comment (MMachadoUofT) [reply src/main/java/entityLayer/CommonTimeBlockManager.java:50]: The TimeBlocks have methods to see whether they are contained within/overlap with other TimeBlocks, so I suspected that we'd simply be using those instead. As I last remember from how we discussed it, the availability is a manager with only 7 timeblocks (one for each day of the week), so doing this shouldn't be too difficult, I don't think.

**Review by Eggathin — APPROVED** [review]

**Review by dadler83 — APPROVED** [review]

**Comment by od-obas1187** [comment]
Looks good so far!

**Review by od-obas1187 — APPROVED** [review]

**Review by bmguiler — COMMENTED** [review]
looks really great thanks for getting this done so quickly

**Pull request merged**